### PR TITLE
Diff SQL handles escape sequences

### PR DESCRIPTION
**Pull request by andrew-wm-arthur**

### CODE REVIEW

File: bats/sql-diff.bats [hunk 1]
```diff
@@ -625,20 +622,40 @@ SQL
...
+    dolt add .
+    dolt commit -m "created table"
+    dolt branch other
+
     dolt sql -q "insert into test (pk, c1) values (0, '\\\\')";
     dolt sql -q	"insert into test (pk, c1) values (1, 'this string ends in backslash\\\\')";
-    dolt diff --sql > $BATS_TMPDIR/input-$$.sql
-    run dolt sql < $BATS_TMPDIR/input-$$.sql
-    skip "backslashes at the end of strings not supported correctly by sql diff"
+    dolt sql -q	"insert into test (pk, c1) values (2, 'this string has \\\"double quotes\\\" in it')";
```
Comment (timsehn) [inline bats/sql-diff.bats:640]: what up w/ the large spaces

File: bats/sql-diff.bats [hunk 2]
```diff
@@ -625,20 +622,40 @@ SQL
...
+    dolt sql -q	"insert into test (pk, c1) values (3, 'it\\'s a contraction y\\'all')";
+    dolt sql -q	"insert into test (pk, c1) values (4, 'backspace \\\b')";
+    dolt sql -q	"insert into test (pk, c1) values (5, 'newline \\\n')";
+    dolt sql -q	"insert into test (pk, c1) values (6, 'carriage return \\\r')";
+    dolt sql -q	"insert into test (pk, c1) values (7, 'tab \\\t')";
+    dolt sql -q	"insert into test (pk, c1) values (8, 'ASCII 26 (Control+Z) \\Z')";
+    dolt sql -q	"insert into test (pk, c1) values (9, 'percent \\%')";
+    dolt sql -q	"insert into test (pk, c1) values (10,'underscore \\_')";
```
Comment (timsehn) [inline bats/sql-diff.bats:648]: make one end in a quote. that's usually trouble

Comment (timsehn) [reply bats/sql-diff.bats:648]: Or '""' or '\"'

File: bats/sql-diff.bats (hunk 1 above)
Comment (andrew-wm-arthur) [reply bats/sql-diff.bats:640]: hmm, I think this is a github artifact

Comment (andrew-wm-arthur) [reply bats/sql-diff.bats:640]: nvm, fixed it

**Pull request merged**